autogpt_platform/backend/backend/notifications/notifications.py

```diff
@@ -145,44 +139,50 @@ def _parse_message(self, message: str) -> NotificationEvent | None:
             logger.error(f"Error parsing message due to non matching schema {e}")
             return None
 
-    async def _process_immediate(self, message: str) -> bool:
+    def _process_immediate(self, message: str) -> bool:
         """Process a single notification immediately, returning whether to put into the failed queue"""
         try:
             parsed = self._parse_message(message)
             if not parsed:
                 return False
             event = parsed.event
             model = parsed.model
-            user_email = get_db_client().get_user_email_by_id(event.user_id)
-            should_send = await self._should_email_user_based_on_preference(
+
+            if event.recipient_email:
```

Comment:
Why are we doing this? What’s the use case here rather than looking up by their id

Reply:
I want to send an email to the system admin not the user triggering the event.